src/coreclr/jit/emitarm64.cpp

```diff
@@ -1058,12 +1058,50 @@ void emitter::emitInsSanityCheck(instrDesc* id)
         // Scalable, 4 regs, to predicate register.
         case IF_SVE_CX_4A: // ........xx.mmmmm ...gggnnnnn.DDDD -- SVE integer compare vectors
             elemsize = id->idOpSize();
-            assert(insOptsScalableStandard(id->idInsOpt())); // xx
-            assert(isPredicateRegister(id->idReg1()));       // DDDD
-            assert(isLowPredicateRegister(id->idReg2()));    // ggg
-            assert(isVectorRegister(id->idReg3()));          // mmmmm
-            assert(isVectorRegister(id->idReg4()));          // nnnnn
             assert(isScalableVectorSize(elemsize));
+            assert(insOptsScalableSimple(id->idInsOpt())); // xx
+            assert(isPredicateRegister(id->idReg1()));     // DDDD
+            assert(isLowPredicateRegister(id->idReg2()));  // ggg
+            assert(isVectorRegister(id->idReg3()));        // nnnnn
+            assert(isVectorRegister(id->idReg4()));        // mmmmm
+            break;
+
+        case IF_SVE_CX_4A_A: // ........xx.mmmmm ...gggnnnnn.DDDD -- SVE integer compare vectors
+            elemsize = id->idOpSize();
+            assert(isScalableVectorSize(elemsize));
+            assert(insOptsScalableWide(id->idInsOpt()));  // xx
+            assert(isPredicateRegister(id->idReg1()));    // DDDD
+            assert(isLowPredicateRegister(id->idReg2())); // ggg
+            assert(isVectorRegister(id->idReg3()));       // mmmmm
+            assert(isVectorRegister(id->idReg4()));       // nnnnn
+            if (id->idInsFmt() == IF_SVE_CX_4A)
```

Comment:
`If` not required - this code only runs for `IF_SVE_CX_4A_A`, and your already doing the assert above.

Reply:
Done

Reply:
should be possible to add this logic in the arm64encoding tool I think.